File: src/nostr/limits.rs
use serde::{Deserialize, Serialize};

#[derive(Debug, Clone, Copy, Serialize, Deserialize)]
pub struct Limits {
    pub max_content_length: usize,
    pub max_subscriptions_per_connection: usize,
}

impl Limits {
    pub fn new(max_content_length: usize, max_subscriptions_per_connection: usize) -> Self {
        Self {
            max_content_length,
            max_subscriptions_per_connection,
        }
    }
}

impl Default for Limits {
    fn default() -> Self {
        Self {
            max_content_length: 65536,
            max_subscriptions_per_connection: 100,
        }
    }
}

#[cfg(test)]
#[path = "limits_test.rs"]
mod limits_test;

File: src/nostr/limits.ts
export interface Limits {
  maxContentLength: number;
  maxSubscriptionsPerConnection: number;
}

export const DEFAULT_LIMITS: Limits = {
  maxContentLength: 65536,
  maxSubscriptionsPerConnection: 100,
};

export function createLimits(maxContentLength: number, maxSubscriptionsPerConnection: number): Limits {
  return { maxContentLength, maxSubscriptionsPerConnection };
}

export const NWC_KINDS = new Set([5, 13194, 23194, 23195, 23196, 23197]);

export function isNwcKind(k: number): boolean {
  return NWC_KINDS.has(k);
}
